nornir/plugins/inventory/netbox.py

```diff
@@ -95,10 +88,45 @@ def __init__(
                 host["data"]["model"] = d["device_type"]
                 host["platform"] = d["platform"]
 
+            # Add NAPALM connection options if they exist for the host platform
+            if napalm_args.get(host["platform"]):
+                host["connection_options"]["napalm"] = {
+                    "extras": {
+                        "optional_args": napalm_args.get(host["platform"], {})
+                        if d["platform"]
+                        else {}
+                    }
+                }
+
             # Assign temporary dict to outer dict
             # Netbox allows devices to be unnamed, but the Nornir model does not allow this
             # If a device is unnamed we will set the name to the id of the device in netbox
             hosts[d.get("name") or d.get("id")] = host
 
         # Pass the data back to the parent class
         super().__init__(hosts=hosts, groups={}, defaults={}, **kwargs)
+
+    def _get(
```

Comment:
Can we rename this method to `_get_resources()`. This would better align with some work that is currently pending.

Reply:
Done